R/correlation.R

```diff
@@ -0,0 +1,181 @@
+INTS <- c(
+  'rtss',
+  'mda',
+  'smc',
+  'tbv',
+  'bednets',
+  'spraying'
+)
+
+#' Class: Correlation parameters
+#' Describes an event in the simulation
+CorrelationParameters <- R6::R6Class(
+  'CorrelationParameters',
+  private = list(
+    interventions = list(),
+    n_ints = function() length(private$interventions),
+    population = 0,
+    rho_matrix = NULL,
+    rho = function() diag(private$rho_matrix),
+    .sigma = NULL,
+    .mvnorm = NULL
+  ),
+  public = list(
+
+    #' @description initialise correlation parameters
+    #' @param parameters model parameters
+    initialize = function(parameters) {
+      # Find a list of enabled interventions
+      enabled <- vlapply(INTS, function(name) parameters[[name]])
+      private$interventions <- INTS[enabled]
+
+      # Initialise a rho matrix for our interventions
+      n_ints <- private$n_ints()
+      private$rho_matrix <- matrix(
+        0,
+        nrow = n_ints,
+        ncol = n_ints,
+        dimnames = list(private$interventions, private$interventions)
+      )
+
+      # Store population for mvnorm draws
+      private$population <- parameters$human_population
+    },
+
+    #' @description Add rho between rounds
+    #' @param int string representing the intervention to update
+    #' @param rho value between 0 and 1 representing the correlation between rounds of
```

Comment:
That's a cool feature. It would need some remodelling though. [The sigma calculation](https://github.com/mrc-ide/malariasimulation/runs/1412848064?check_suite_focus=true) wouldn't work in this model.

```r
private$.sigma <- sqrt(rho / (1 - rho))
```

Reply:
Yeah, agreed. I wouldn't prioritise this (no one has been clamouring for it as far as I know!)